File: app/src/main/java/ru/radiationx/anilibria/mvp/BasePresenter.java
package ru.radiationx.anilibria.mvp;

/**
 * Created by radiationx on 05.11.17.
 */

public class BasePresenter<ViewT> implements IBasePresenter<ViewT> {
    protected ViewT view;

    public BasePresenter(ViewT view) {
        this.view = view;
    }

    @Override
    public void onResume(ViewT view) {
        this.view = view;
    }

    @Override
    public void onPause() {
        this.view = null;
    }
}

File: app-mobile/src/main/java/ru/radiationx/anilibria/presentation/common/BasePresenter.kt
package ru.radiationx.anilibria.presentation.common

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob
import kotlinx.coroutines.cancel
import moxy.MvpPresenter
import moxy.MvpView
import ru.terrakok.cicerone.Router

/**
 * Created by radiationx on 05.11.17.
 */


@Deprecated("use viewmodel", level = DeprecationLevel.ERROR)
open class BasePresenter<ViewT : MvpView>(private val router: Router) : MvpPresenter<ViewT>() {

    val viewModelScope by lazy {
        CoroutineScope(Dispatchers.Main.immediate + SupervisorJob())
    }

    override fun onDestroy() {
        viewModelScope.cancel()
    }

    fun onBackPressed() {
        router.exit()
    }
}
